stdlib/email/src/main/java/org/ballerinalang/stdlib/email/server/EmailListenerHelper.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.stdlib.email.server;
+
+import org.ballerinalang.jvm.BRuntime;
+import org.ballerinalang.jvm.values.MapValue;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.jvm.values.api.BString;
+import org.ballerinalang.stdlib.email.util.EmailConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Helper class for listener functions.
+ *
+ * @since 1.3.0
+ */
+public class EmailListenerHelper {
+
+    private EmailListenerHelper() {
+        // private constructor
+    }
+
+    /**
+     * Initialize a new EmailConnector for the listener.
+     * @param emailListener Listener that places emails in Ballerina runtime
+     * @param serviceEndpointConfig Email server endpoint configuration
+     * @throws EmailConnectorException If the given protocol is invalid
+     */
+    public static void init(ObjectValue emailListener, MapValue<Object, Object> serviceEndpointConfig)
+            throws EmailConnectorException {
+        final EmailListener listener = new EmailListener(BRuntime.getCurrentRuntime());
+        Map<String, Object> paramMap = getServerConnectorParamMap(serviceEndpointConfig);
+        EmailConnector emailConnector = EmailConnectorFactory.createServerConnector(paramMap, listener);
+        emailListener.addNativeData(EmailConstants.EMAIL_SERVER_CONNECTOR, emailConnector);
+    }
+
+    /**
+     * Register a new service for the listener.
+     * @param emailListener Ballerina email listener
+     * @param service Ballerina service to be listened
+     * @param serviceName Name of the Ballerina service to be listened
+     */
+    public static void register(ObjectValue emailListener, ObjectValue service, BString serviceName) {
+        EmailConnector emailConnector = (EmailConnector) emailListener.getNativeData(
+                EmailConstants.EMAIL_SERVER_CONNECTOR);
+        EmailListener listener = emailConnector.getEmailListener();
+        listener.addService(service, serviceName.getValue());
```

Comment:
```suggestion
        listener.addService(service, service.getType().getName());
```

Reply:
I doubt we get the correct service name from `serviceName` param. better we could get it from `service.getType().getName()`

Reply:
Used this `serviceName` and removed UUID based service map keys.